cloudinit/stages.py

```diff
@@ -1101,11 +1102,14 @@ def should_run_on_boot_event():
                 log_details=False,  # May have wifi passwords in net cfg
                 log_deprecations=True,
             )
-        # ensure all physical devices in config are present
-        self.distro.networking.wait_for_physdevs(netcfg)
 
-        # apply renames from config
-        self._apply_netcfg_names(netcfg)
+        if not isinstance(self.distro, Ubuntu):
+
+            # ensure all physical devices in config are present
+            self.distro.networking.wait_for_physdevs(netcfg)
+
+            # apply renames from config
+            self._apply_netcfg_names(netcfg)
```

Comment:
I think this commit needs a rebase on upstream/main to avoid pulling in your unapproved PR.